File: lib/models/todos.dart
class Todo {
  String task;
  String description;
  String date;
  int priority;
  int id;
  bool completed;

  Todo({
    this.task,
    this.description,
    this.date,
    this.priority,
    this.id,
    this.completed,
  });

  Todo.fromJson(Map<String, dynamic> json) {
    task = json['task'];
    description = json['description'];
    date = json['date'];
    priority = json['priority'];
    id = json['id'];
    completed = json['completed'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['task'] = this.task;
    data['description'] = this.description;
    data['date'] = this.date;
    data['priority'] = this.priority;
    data['id'] = this.id;
    data['completed'] = this.completed;
    return data;
  }
}

File: server/api/utils/todos.py
from sqlalchemy.orm import Session

# Models used to access data
from db.models.todos import Todos
# Schemas used to validate incoming data
from pydantic_schemas.todos import TodoBase


def get_task(db: Session, task_id: int):
    return db.query(Todos).filter(Todos.id == task_id).first()


def get_all_tasks(db: Session):
    return db.query(Todos).all()


def create_task(db: Session, task: TodoBase):
    new_task = Todos(
        task=task.task,
        description=task.description,
        date=task.date,
        priority=task.priority
    )
    db.add(new_task)
    db.commit()
    db.refresh(new_task)
    return new_task


def delete_task(db: Session, task_id: int):
    task = get_task(db, task_id)
    db.delete(task)
    db.commit()
    return task


def complete_task(db: Session, task_id: int):
    todo = get_task(db, task_id)
    todo.completed = not todo.completed
    db.commit()
    db.refresh(todo)
    return todo
